app/Http/Middleware/RequireTwoFactorAuthentication.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Pterodactyl\Http\Middleware;
+
+use Closure;
+use Krucas\Settings\Settings;
+use Prologue\Alerts\AlertsMessageBag;
+
+class RequireTwoFactorAuthentication
+{
+    const NOBODY = 0;
+    const ADMINISTRATORS = 1;
+    const USERS = 2;
+    const EVERYBODY = 3;
+
+    /**
+     * @var \Prologue\Alerts\AlertsMessageBag
+     */
+    protected $alert;
+
+    /**
+     * @var \Krucas\Settings\Settings
+     */
+    protected $settings;
+
+    /**
+     * RequireTwoFactorAuthentication constructor.
+     *
+     * @param \Prologue\Alerts\AlertsMessageBag $alert
+     * @param \Krucas\Settings\Settings         $settings
+     */
+    public function __construct(Settings $settings, AlertsMessageBag $alert)
```

Comment:
This is undocumented, so my fault here, but I've been using an Alphabetical scheme for constructor classes and ordering the protected/public variables. So this should be `__construct(AlertsMessageBag $alert, Settings $settings)` and then `$this->alert` before settings.

Reply:
Not a problem.